spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -407,14 +422,35 @@ public Scan build() {
 
   private Scan buildBatchScan() {
     Schema expectedSchema = schemaWithMetadataColumns();
+    org.apache.iceberg.Scan scan =
+        buildIcebergBatchScan(false /* not include Column Stats */, expectedSchema);
+    if (pushedLimit > 0 && hasDeletes(scan)) {
+      LOG.info("Skipping limit pushdown: detected row level deletes");
+      pushedLimit = -1;
+    }
```

Comment:
[Q] can we push down limit even when some data filters are not completely pushed down to scan layer ?

Reply:
If data filters are not completely pushed down to scan layer, Spark will not call `pushLimit(int limit)`. I will add some negative tests for the filter case.

Reply:
Interesting read more about it, thanks for introducing this feature in spark in the first place https://github.com/apache/spark/pull/34291/files, wondering if we should also override this `isPartiallyPushed` as well

Reply:
I think we want `isPartiallyPushed` to be true so Spark will do a global limit

Reply:
Was wondering if the filter is partition filter or no filter at all and we know the amount of files we are gonna scan and have enough confidence that we will get those number of records can we set this to false then, hinting spark that global limit is not required, though it will be a very niche use case and maybe not worth it

Reply:
I think we can't skip the global limit unless we are certain that there is only one Spark task.